### DISCUSSION TITLE
Adding/Updating Online Meeting Issue

### DISCUSSION THREAD

**Discussion by Grateful-Viv** [code4recovery/12-step-meeting-list#833] [Q&A]
Using TSML v3.14.4

I was testing out the columns on a staging site (see #806) and the issue became apparent when I tried to change the location name of an online meeting. The attendance-option was "tc". Below is a screen shot of what it looked like before any changes were done.

![Edit-Meeting-“Wake-Up-Call-24-Hours-Day-Book”-‹-Southern-Delaware-Intergroup-Staging-Site-—-WordPress](https://user-images.githubusercontent.com/62247421/174598101-b886757d-28d0-4679-8c98-496052d5049f.jpg)

Then I changed Location Name and used an approximate address. See below.

![2-Edit-Meeting-“Wake-Up-Call-24-Hours-Day-Book”-‹-Southern-Delaware-Intergroup-Staging-Site-—-WordPress](https://user-images.githubusercontent.com/62247421/174598983-d3f8724d-8c6a-4e05-8c0c-99b6870bcf48.png)

After Update, the location name and address reverted back to what it originally and set "Yes" to Can I currently attend this meeting in person?

So I tried to edit another meeting on another staging site that didn't have this added to the wpconfig.php file. Changed location name to something unique like Early Birds Online and used Bethany Beach, DE as the address. 

![Edit-Meeting-“Early-Birds-Grapevine”-‹-Southern-Delaware-Intergroup-Staging-Site-—-WordPress](https://user-images.githubusercontent.com/62247421/174603069-b304ea87-1171-444d-a35e-4d96e9aa0271.png)

And it did the same thing. Reverted back to it's former address and changed "Yes" to Can I currently attend this meeting in person?

Then I tried to add a new test meeting. And even though it published, it removed the location, address and region. And set "Yes" to Can I currently attend this meeting in person?

![Edit-Meeting-“Online-Meeting-Test”-‹-SDIAA-Staging-TWO-—-WordPress](https://user-images.githubusercontent.com/62247421/174601171-89ddb9c6-0228-4731-97cc-a6854ee6a859.png)

The weird thing is it shows up on the list of meetings as an online meeting. And when I edit, even though the update is displaying it incorrectly, the data displayed is also showing it correctly. And this is true on either staging site.

![Meetings](https://user-images.githubusercontent.com/62247421/174602284-7f1d1534-9ee2-4aff-9a16-97611d2c2526.png)

Update: After a few edits, it is now showing these meetings as hybrid. :(

I downloaded the CSV, and it is retaining the address of the location.

**Comment by tech2serve** [comment]
First, I apologize that nobody has replied to you yet. 

I do confess I'm having trouble understanding the issue, or the question. 

Since I know more work has been done to the code regarding display of the location name, are you still experiencing the issue with the latest version of TSML?

If so, it might help to just pick one meeting that is giving you fits, open a javascript/debug console, and try making a screen recording of the steps you're taking, to include clicking on the debugging variable in the console. 

Were these meetings previously in-person and now you're effectively trying to take them to purely online?

**Reply by Grateful-Viv** [reply]
These meetings were in person, now online with TC as the meeting attendance option 

I’m using the plugin with TSML-UI as front end, but back end is the legacy version. 

Location, Address & Region seem intertwined. If I change “Online” to “Online Meeting”, all instances where the location was “Online” will change to the new name. This is also true if addressees are changed as well as region. I can’t have a location “Online Meeting” for different regions or addresses. It’s maddening. And when I would change the status by removing the original street address or location - it would revert back to the original location and change meeting online to in person. 

I resolved this weird thing by uploading the csv which I changed manually. I still couldn’t have Online Meeting in Lewes and also have Online Meeting in Rehoboth. It was one or the other. But at least it didn’t jump back to setting address back and making it in person. 

Does any of this additional info help?

**Reply by tech2serve** [reply]
Yes. Although I do believe a screen recording with the browser's console visible might help to see what is going on.

I need to have an internal discussion before I publicly hypothesize on what I think might be happening.

**Reply by Grateful-Viv** [reply]
how do I make a screen recording of the JavaScript console in Windows?

**Reply by tech2serve** [reply]
I can't help you much with that as I don't do windows. Get it. Hahahaha. I crack myself up.

On a Mac there are tools built in that can be used for screen recordings, and all one would have to do is make the console visible and capture the entire window. 

Here is [one article](https://support.microsoft.com/en-us/office/create-screen-recording-d4c066c1-d517-44ce-8bd4-65314c97aa40) on Microsoft's website. I can say it isn't accurate about needing to use Chrome on a Mac, but it might be good Information on a PC.

**Reply by brianw-area24** [reply]
For the location names, what about using names such as "Online - <region name>"? This way you'd have a separate name for each region.

**Reply by Grateful-Viv** [reply]
>For the location names, what about using names such as "Online - "? This way you'd have a separate name for each region.

@brianw-area24, I ended up doing that, but it's more than changing the location name. The combination of Location, address and region need to be unique. Otherwise it will overwrite the existing data.

Examples:
Location: "Online - Rehoboth" Address: "Rehoboth Beach DE" Region: "Rehoboth"
Location: "Online - Lewes" Address: "Lewes DE" Region: "Lewes"
Location: "Online Meeting". Address: "Sussex County DE" Region: "Sussex County"
Location: "Wake Up Call Online" Address: "Harbeson DE" Region: "Harbeson"
Location: "Early Birds Online" Address: "Bethany Beach DE" Region: "Bethany"

If I change the location name of one of the meetings, it will affect all meetings for that address and region. Same is true if I change the address, it will change all meetings with that location name to the new address. It's kind of maddening, and for most of these online meetings, I just let it go with Online Meeting and Sussex County DE and called it a day.

**Comment by brianw-area24** [comment]
If I'm understanding this correctly, it's changing the "Can I currently attend this meeting in person?" answer to Yes, when you edit meetings. I've imported your meetings into my demo site, and I'm not able to recreate the problem.

I'm curious if you can try this using a different browser, especially if it can be from a different computer? I'm wondering if there is something in the browser that's auto filling entries in the form.

**Comment by lionssharedigital** [comment]
Both Austin AA and Nashville are having issues with plugin. They are unable to add new or update meetings at all. We see nothing in the Javascript console. Here's a Loom:
https://www.loom.com/share/d48247828bd74e4d8892e136bcaeb889

I am not seeing any meaningful issues in Console. 

Firefox issue with Console: https://capture.dropbox.com/sZ0lKVtMlvLsQAlL

From Bill with Nashville. Screenshots attached.
The problem is that I still can't create a new meeting as it will not let me add an address and therefore am unable to publish.  I sent screen shots.  In the first, you see that I typed in Church St. address and nothing.  In the second, the publish button is not highlighted.  In the third, I used an embedded location from other meetings in town, and the address registers and a map is provided with correct location.  In the fourth screen shot, the publish button is highlighted  and I am able to publish meeting.  This is the problem needing corrected.

![2022-07-13 (8)](https://user-images.githubusercontent.com/10728016/178809486-96dc7eca-cda7-4f35-9494-821b189eb3ca.png)
![2022-07-13 (7)](https://user-images.githubusercontent.com/10728016/178809498-adf62f7d-c387-47a7-8e7c-3bc662523206.png)
![2022-07-13 (6)](https://user-images.githubusercontent.com/10728016/178809505-89744bf0-dad0-494a-b242-18e383f7accf.png)
![2022-07-13 (5)](https://user-images.githubusercontent.com/10728016/178809511-8f24a65d-7b6a-4877-962b-ab7f76829ec1.png)

**Reply by Grateful-Viv** [reply]
@lionssharedigital, I added the same addresses as you did on my staging site -- the one on 23rd Street and the one on Church Street after an edit. In both cases the meeting published and updated, but the weird part is when I went to edit, the location and address was blank until I added a region.

Also, don't know if this is related, but I had caching issues where updates were not showing. Though I still had enabled Dynamic Caching, I ended up excluding` /meetings/* `from being cached.

And this may be relevant - I am using the Beta Gateway under Address Geocoding.

**Reply by lionssharedigital** [reply]
Got it. That is a bit helpful. I am still having problems. I think our issue is Mapbox and Google Maps involved. I've added a Google Map API key and it's working better now. I think you are right that you have to select a region. 

Things are working for Nashville now that I'm using Google API Key. I'll try this with Austin AA and I'll see if that fixes his issues as well.

**Reply by lionssharedigital** [reply]
Yes, setting AustinAA.org to use Google API Key and Beta - API Gateway fixed the isssue. Removing the Mapbox token and BETA - API Gateway allowed us to edit the content. Then I needed an Google API Key to get the map to work on the front end.

**Reply by tim-rohrer** [reply]
We were experiencing an issue with our Google account (sigh....) which has now been resolved.

You should be able to return to:

MapBox token for mapping (or you can continue to use your own Google mapping key if you prefer).
Legacy for geocoding (or you can use the Beta, but it will be replaced in the (not too distant, perhaps) future).

Apologies for the interruption of service. I know it is frustrating.
